### PR TITLE
fix(dockerfile): add yarn

### PR DESCRIPTION
**Pull request by arnulfojr**
Install the latest yarn version in the docker image.
Fixes https://github.com/aws/aws-cdk/issues/4843
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -22,7 +22,10 @@ RUN yum -y --security update \
   && grep " node-v$NODE_VERSION-linux-x64.tar.xz\$" SHASUMS256.txt | sha256sum -c - \
   && tar -xJf "node-v$NODE_VERSION-linux-x64.tar.xz" -C /usr/local --strip-components=1 --no-same-owner \
   && ln -s /usr/local/bin/node /usr/local/bin/nodejs \
-  && rm "node-v$NODE_VERSION-linux-x64.tar.xz" SHASUMS256.txt
+  && rm "node-v$NODE_VERSION-linux-x64.tar.xz" SHASUMS256.txt \
+  && curl -fsSLO "https://yarnpkg.com/latest.tar.gz" \
```
Comment (nmussy) [inline Dockerfile]: We probably should add a hash verification, like is done for done above. See [docker-node Dockerfile](https://github.com/nodejs/docker-node/blob/70431ea2d28f19d47dbadd7bee927fbf942672d3/10/jessie/Dockerfile#L45-L62) as an example

Comment (arnulfojr) [reply Dockerfile]: Lastest version is ok or shall I fix the yarn version @nmussy ?

Comment (arnulfojr) [reply Dockerfile]: Fixed the version to `1.19.1` let me know if you'd like to change it otherwise :)

Comment (nmussy) [reply Dockerfile]: I haven't tested your Dockerfile, but it seems you're missing the `ln` commands required to add `yarn` to the path (see [example](https://github.com/nodejs/docker-node/blob/70431ea2d28f19d47dbadd7bee927fbf942672d3/10/jessie/Dockerfile#L60-L61)).

Comment (arnulfojr) [reply Dockerfile]: Well, it depends on how you want to save the yarn binary but atm the yarn is actually exported in the path becuase is extracted directly into `/usr/local`
In the current PR I used the same extraction command than the one used for node,

```sh
tar zvxf yarn-v$YARN_VERSION.tar.gz -C /usr/local --strip-components=1 --no-same-owner
```

That means that the first component from `yarn-v<numbers>` will be used to save the file -> `yarn`.
There's also the possibility of using the symlink while preserving the version number ->
```sh
&& tar zvxf yarn-v$YARN_VERSION.tar.gz -C /opt/ --no-same-owner \
&& ln -s /opt/yarn-v$YARN_VERSION/bin/yarn /usr/local/bin/yarn
```

I chose the first one just to be consistent with the node extraction but if you think the latter is better (more clear) then we can go for it

Comment (nmussy) [reply Dockerfile]: My bad, I missed the `--strip-components=1`. 😄

**Pull request merged**